File: BotBasedOtouto/V2/Spotify.lua
-- Spotify Plugin for bot based on otouto
-- ByTiagoDanin - Telegram.me/tiagodanin

local PLUGIN = {}

PLUGIN.doc = [[
	/spotify <music>
	Track Spotify byTiagoDanin.
]]

PLUGIN.triggers = {
	'^/spotify'
}

function PLUGIN.action(msg)

	local input = get_input(msg.text)
	if not input then
		return send_msg(msg, PLUGIN.doc)
	end

	local BASE_URL = "https://api.spotify.com/v1/search"
	local URLP = "?q=".. (URL.escape(input) or "").."&type=track&limit=5" -- Limit 5

	local decj, tim = HTTPS.request(BASE_URL..URLP)
	if tim ~=200 then return nil  end

	local spotify = JSON.decode(decj)
	local tables = {}
	for pri,result in ipairs(spotify.tracks.items) do
	table.insert(tables, {
		spotify.tracks.total,
		result.name,
		result.external_urls.spotify
	})
	end

	local gets = ""
	for pri,cont in ipairs(tables) do
		gets=gets.."▶️ "..cont[2].." - "..cont[3].."\n"
	end

	local text_end = gets -- Text END
	if gets == "" then
		text_end = "Not found music"
	end

	send_message(msg.chat.id, text_end)
end

return PLUGIN

File: BotBasedPolaris/Spotify.py
# -*- coding: utf-8 -*-
# Spotify Plugin for bot based on Polaris
# ByTiagoDanin - Telegram.me/tiagodanin

from __main__ import *
from utils import *

commands = [
	'^spotify',
	'^spo',
	'^s'
]

parameters = {('query', True)}

description = 'Returns results.\nAlias: /s, /spo'
action = 'typing'


def run(msg):
	input = get_input(msg['text'])

	if not input:
		doc = get_doc(commands, parameters, description)
		return send_message(msg['chat']['id'], doc, parse_mode="Markdown")

	url = 'https://api.spotify.com/v1/search'
	params = {
		'q': input,
		'type': 'track',
		'limit': '4'
	}

	jstr = requests.get(
	url,
		params=params,
	)

	if jstr.status_code != 200:
		return send_message(msg['chat']['id'], 'Erro connection.')

	spotify = json.loads(jstr.text)
	if spotify['tracks']['total'] == 0:
		return send_message(msg['chat']['id'], 'No results found.')

	text = '*Spotify Search*: "_' + input + '_"\n\n'
	for i in range(0, len(spotify['tracks']['items'])):
		title = spotify['tracks']['items'][i]['name'] +  ' - ' + spotify['tracks']['items'][i]['album']['name']
		if len(str(title)) > 48:
			title = title[:44] +  '...'
			stitle = title.replace('[', ' |').replace(']', '| ').replace('(', '').replace(')', '')
			surl = spotify['tracks']['items'][i]['external_urls']['spotify']
		text += U'• [' + stitle + '](' + surl + ')\n\n'

		send_message(msg['chat']['id'], text, parse_mode="Markdown")
